src/state/slices/portfolioSlice/portfolioSlice.ts

```diff
@@ -31,13 +21,28 @@ export const portfolio = createSlice({
       state.accounts.byId = { ...state.accounts.byId, ...payload.accounts.byId }
       const accountIds = Array.from(new Set([...state.accounts.ids, ...payload.accounts.ids]))
       state.accounts.ids = accountIds
-
-      state.assetBalances.byId = mergeWith(
-        state.assetBalances.byId,
-        payload.assetBalances.byId,
-        upsertAndSum
-      )
-
+      /**
+       * when fetching an account we got to calculate the difference between the
+       * state accountBalance and the payload accountBalance for each of account assets
+       * and then add [or subtract] the diff to the state.assetBalances related item.
+       */
+      payload.accountBalances.ids.forEach(accountSpecifier => {
+        // iterate over the account assets balances and calculate the diff
+        Object.entries(payload.accountBalances.byId[accountSpecifier]).forEach(
+          ([caip19, newAccountAssetBalance]) => {
+            // in case if getting accounts for the first time
+            const currentAccountBalance = bnOrZero(
+              state.accountBalances.byId[accountSpecifier]?.[caip19]
+            )
+            // diff could be both positive [tx type -> receive] and negative [tx type -> send]
+            const diff = bnOrZero(newAccountAssetBalance).minus(currentAccountBalance)
+            // get current asset balance from the state
+            const currentAssetBalance = bnOrZero(state.assetBalances.byId?.[caip19])
+            // update state.assetBalances with calculated diff
+            state.assetBalances.byId[caip19] = currentAssetBalance.plus(diff).toString()
```

Comment:
Yeah it's verbose, but I'm all for removing cognitive overhead through descriptive variable names:

```suggestion
            // diff could be both positive [tx type -> receive] and negative [tx type -> send]
            const differenceBetweenCurrentAndNew =
              bnOrZero(newAccountAssetBalance).minus(currentAccountBalance)
            // get current asset balance from the state
            const currentAssetBalance = bnOrZero(state.assetBalances.byId?.[caip19])
            // update state.assetBalances with calculated diff
            state.assetBalances.byId[caip19] = currentAssetBalance
              .plus(differenceBetweenCurrentAndNew)
              .toString()
```